### DISCUSSION TITLE
Get TableInfo for a table created using customstatement.

### DISCUSSION THREAD

**Discussion by MadhuprakashK** [simolus3/drift#1744] [Q&A]
I have two DB in my application (Db1 and Db2) Db1 tables are known and are generated using .g files , but Db2 is generated dynamically using a metadata file from asset through custom queries.

The problem I'm facing is during bulk insert , for the DB1 I am able to use batch and bulk insert  which is efficient. But in case of DB2 I had to run a for loop and insert one by one which is taking quite a long time(7-10mins) to process 15k records.

I tried using transactions but it doesn't help.

If we can somehow get the table info list of DB2 , I will be able to use batch.insertAll() or is there a way to execute a list of custom queries using batch?

Any suggestions??

**Comment by simolus3** [comment]
> But in case of DB2 I had to run a for loop and insert one by one which is taking quite a long time(7-10mins) to process 15k records.

I assume you're also using dynamic and custom insert statements for DB2? You can use a batch for those too:

```dart
await batch((b) {
  for (final entry in myList) {
    b.customStatement('INSERT INTO ...', [...]);
  }
});
```

> If we can somehow get the table info list of DB2 , I will be able to use batch.insertAll() or is there a way to execute a list of custom queries using batch?

You can manually write table classes with `class MyTable extends Table with TableInfo<dynamic, MyTable>`, but since their structure is dynamic drift can't help you generate them.

**Comment by KiraKmp** [comment]
await appDb.batch((batch) {
             var index = 1;
             for(var query in queries){
               log("Processing query ${index.toString()}");
               batch.customStatement(query);
               index++;
             }
            });

I'm trying this way i have over 11k queries in the list , but i don't see any effect on DB. No data is getting inserted

**Reply by simolus3** [reply]
Is there any exception being logged? The batch should definitely add a statement when `customStatement` is called. But know that stream queries don't update automatically for `customStatement`.
